src/coins.cpp

```diff
@@ -647,6 +690,26 @@ bool CCoinsViewCache::BatchWrite(CCoinsMap &mapCoins,
     }
     mapSidechainEvents.clear();
 
+    for (CCswNullifiersMap::iterator child_it = cswNullifiers.begin(); child_it != cswNullifiers.end();)
+    {
+        if (child_it->second.flags & CNullifiersCacheEntry::DIRTY) { // Ignore non-dirty entries (optimization).
+            CCswNullifiersMap::iterator parent_it = cswNullifiers.find(child_it->first);
+
+            if (parent_it == cswNullifiers.end()) {
```

Comment:
same here.

Reply:
Honestly, this `for` loop can be simplified a lot.
Do similar to mapSidechains update:
Loop thought `cswNullifiers`. If item flag is Fresh or Erased -> do checks/asserts and update the value in `cacheCswNullifiers` map.
If item flag is `DEFAULT` -> do nothing. `default: assert(false)`